virtualmachine/vsphere/util.go

```diff
@@ -875,6 +875,7 @@ var cloneFromTemplate = func(vm *VM, dcMo *mo.Datacenter, usableDatastores []str
 		config.DeviceChange = append(config.DeviceChange, conf...)
 	}
 
+	checkCustomSpecMutex.Lock()
```

Comment:
Can we have a comment here emphasizing that critical section starts here, to remind unlocking and avoid adding unnecessary code in this section in future.